pkgs/by-name/go/go-blueprint/package.nix

```
@@ -0,0 +1,26 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}:
+
+buildGoModule rec {
+  pname = "go-blueprint";
+  version = "0.8.1";
+
+  src = fetchFromGitHub {
+    owner = "Melkeydev";
+    repo = "go-blueprint";
+    rev = "v${version}";
+    hash = "sha256-lIx95DY8UpxTW5Zra0APseuXdknsdZAohiGFdwkHE4s=";
+  };
+
+  vendorHash = "sha256-WBzToupC1/O70OYHbKk7S73OEe7XRLAAbY5NoLL7xvw=";
+
```

This program supports generating completion.
```suggestion
  nativeBuildInputs = [ installShellFiles ];

  postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
    installShellCompletion --cmd go-blueprint \
      --bash <($out/bin/go-blueprint completion bash) \
      --fish <($out/bin/go-blueprint completion fish) \
      --zsh <($out/bin/go-blueprint completion zsh)
  '';
```

---

Also it's highly recommended to add https://nixos.org/manual/nixpkgs/unstable/#versioncheckhook, if possible.

---

Thank you, added completion and versionCheckHook.